pinot-broker/src/test/java/com/linkedin/pinot/broker/requesthandler/RangeMergeOptimizerTest.java

```diff
@@ -0,0 +1,183 @@
+    actualTree = buildFilterQueryTree("select * from table where time >= 10 and time <= 20 and time <= 15", true);
+    expectedTree = buildFilterQueryTree("select * from table where time between 10 and 15", false);
+    compareTrees(actualTree, expectedTree);
+
+    // Query with multiple predicates
+    actualTree =
+        buildFilterQueryTree("select * from table where time >= 10 and time <= 20 and time <= 15 and foo = 'bar'",
+            true);
+    expectedTree = buildFilterQueryTree("select * from table where time between 10 and 15 and foo = 'bar'", false);
+    compareTrees(actualTree, expectedTree);
+
+    // Query with nested predicates
+    actualTree =
+        buildFilterQueryTree("select * from table where (time >= 10 and time <= 20) and ((time >= 5 and time <= 15))",
+            true);
+    expectedTree = buildFilterQueryTree("select * from table where time between 10 and 15", false);
+    compareTrees(actualTree, expectedTree);
+
+    // Query with nested predicates where not all ranges can be merged
+    actualTree = buildFilterQueryTree(
+        "select * from table where (time >= 10 and time <= 20) and (foo1 = 'bar1' and (foo2 = 'bar2' and (time >= 5 and time <= 15)))",
+        true);
+    expectedTree = buildFilterQueryTree(
+        "select * from table where time between 10 and 20 and (foo1 = 'bar1' and (foo2 = 'bar2' and (time between 5 and 15)))",
+        false);
+    compareTrees(actualTree, expectedTree);
+
+    // Query with time range in different levels of tree and all ranges can be merged
+    actualTree =
+        buildFilterQueryTree("select * from table where (((time >= 10 and time <= 20) and time >= 5) and time <= 15)",
+            true);
+    expectedTree = buildFilterQueryTree("select * from table where time between 10 and 15", false);
+    compareTrees(actualTree, expectedTree);
+
+    // Query with time range of one value.
+    actualTree =
+        buildFilterQueryTree("select * from table where (time > 10 and time <= 20) and (time >= 20 and time <= 30)",
+            true);
+    expectedTree = buildFilterQueryTree("select * from table where time between 20 and 20", false);
+    compareTrees(actualTree, expectedTree);
+
+    // Query with OR predicates
+    actualTree = buildFilterQueryTree("select * from table where (foo = 'bar' or time < 10)", true);
+    expectedTree = buildFilterQueryTree("select * from table where (foo = 'bar' or time < 10)", false);
+    compareTrees(actualTree, expectedTree);
+
+    // Query without time column
+    actualTree = buildFilterQueryTree("select * from table where foo = 'bar'", true);
+    expectedTree = buildFilterQueryTree("select * from table where foo = 'bar'", false);
+    compareTrees(actualTree, expectedTree);
+  }
+
+  /**
+   * Helper method to compare two filter query trees
+   * @param actualTree Actual tree
+   * @param expectedTree Expected tree
+   */
+  private void compareTrees(FilterQueryTree actualTree, FilterQueryTree expectedTree) {
```

Comment:
This part of the test is very similar to the one in `MultipleOrEqualitiesToInClauseFilterQueryTreeOptimizerTest`, might be a good idea to refactor the tests to avoid duplicated code.